### PR TITLE
Support exotic architectures removed in OCaml 5.0

### PR DESCRIPTION
**Pull request by benmandrew**
Support for s390, arm32, x86_32, and ppc64 have not yet been added to OCaml 5.0, but we attempt to build these combinations anyway and fail, leaving the architectures untested. This PR uses a change in `ocaml-version` (https://github.com/ocurrent/ocaml-version/pull/61) to choose the latest version of OCaml that is supported by a given architecture.

This cannot be merged until the `ocaml-version` change is merged and released.

Fixes https://github.com/ocurrent/ocaml-ci/issues/756.

### CODE REVIEW

**Comment by tmcgilchrist** [comment]
@benmandrew re-vendor `ocaml-version` and try this out on ocaml-ci against [https://github.com/mirage/mirage-crypto](https://github.com/mirage/mirage-crypto/)

**Review by kit-ty-kate — CHANGES_REQUESTED** [review]
OCaml-CI should instead test 4.14 and 5.0 as if they were both latest versions, given 4.14 is LTS. This is what opam-repo-ci does for example (see https://github.com/ocurrent/opam-repo-ci/pull/196)

**Comment by benmandrew** [comment]
I've vendored ocaml-version pointing to my own fork.

Basic question, but if the changes are OCaml-CI specific then what's the standard way to integrate a submodule in? Pointing to my fork isn't a permanent solution, but it also seems strange to maintain a separate branch on [`ocurrent/ocaml-version`](https://github.com/ocurrent/ocaml-version) specifically for usage in OCaml-CI.

File: service/conf.ml [hunk 1]
```diff
@@ -52,7 +52,9 @@ end
 module OV = Ocaml_version
 module DD = Dockerfile_opam.Distro
 
-let default_compiler = OV.(Releases.latest |> without_patch)
+let default_compilers =
+  OV.(List.map with_just_major_and_minor Releases.[ v4_14; v5_0 ])
```
Comment (kit-ty-kate) [inline service/conf.ml:56]:
```suggestion
  OV.(List.map with_just_major_and_minor Releases.[ v4_14; latest ])
```

**Comment by kit-ty-kate** [comment]
I don't understand why this fork is necessary

**Comment by kit-ty-kate** [comment]
> OCaml-CI assumes that all architectures work fully on 5.0+. It tries to build these architectures with the master distro and latest version, which fails at some point leading to those arches not being tested. 

They all do work though, only in bytecode-only mode (which is a valid state of the OCaml compiler).

The issue in https://github.com/ocurrent/ocaml-ci/issues/756 is just that when a package depends on `num` then it won't be tested due to `num` being unavailable in bytecode-only mode. So simply testing both 4.14 and 5.0 fixes this issue and the ocaml-version fork is not needed.

**Comment by benmandrew** [comment]
Ahh I understand what you mean, thanks for the explanation.

**Review by kit-ty-kate — APPROVED** [review]

**Pull request merged**